app/journey/Journey.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package journey
+
+import java.time.LocalDate
+
+import play.api.libs.json.{Json, OFormat}
+import ssttpcalculator.CalculatorPaymentScheduleExt
+import timetopaycalculator.cor.model.{CalculatorInput, PaymentSchedule}
+import timetopaytaxpayer.cor.model.Taxpayer
+import uk.gov.hmrc.selfservicetimetopay.models._
+
+final case class Journey(
+    _id:                    JourneyId,
+    maybeAmount:            Option[BigDecimal]                   = None,
+    schedule:               Option[CalculatorPaymentScheduleExt] = None,
+    bankDetails:            Option[BankDetails]                  = None,
+    existingDDBanks:        Option[DirectDebitBank]              = None,
+    maybeTaxpayer:          Option[Taxpayer]                     = None,
+    maybeCalculatorData:    Option[CalculatorInput]              = None,
+    durationMonths:         Int                                  = 2,
+    maybeEligibilityStatus: Option[EligibilityStatus]            = None,
+    debitDate:              Option[LocalDate]                    = None,
+    ddRef:                  Option[String]                       = None) {
+
+  def amount: BigDecimal = maybeAmount.getOrElse(throw new RuntimeException(s"Expected 'amount' to be there but was not found. [${_id}] [${this}]"))
```

Comment:
The problem here is that the `amount` is not always given. It will throw exception if you try to get it if it's not there (which is our purpose). 
If you make it eager it will throw exceptions always (for example in the beginning of the journey, where the journey is created)